### PR TITLE
Update schema of DataSource in v3 YAML

### PR DESCRIPTION
**Pull request by jamevaalet**
Updating the YAML v3 schema for DataSource according to below [tech design](https://microsoft.sharepoint-df.com/:fl:/g/contentstorage/CSP_fbfb0a53-e4e1-4407-8b36-716d46cc0d5f/EYaglxwWl0FLiP_NCzeYtJ4BKak3hM7gZMwd0rxfUa4CVg?e=HeWS9L&nav=cz0lMkZjb250ZW50c3RvcmFnZSUyRkNTUF9mYmZiMGE1My1lNGUxLTQ0MDctOGIzNi03MTZkNDZjYzBkNWYmZD1iJTIxVXdyNy0tSGtCMFNMTm5GdFJzd05YeS1obEdfdlNoZElyU200RG5YejJMcVpLTE5yMVZmRFJJVFV3U2dHNjBxYyZmPTAxRjNWVERBNEdVQ0xSWUZVWElGRllSNzZOQk0zWlJORTYmYz0lMkYmYT1Mb29wQXBwJnA9JTQwZmx1aWR4JTJGbG9vcC1wYWdlLWNvbnRhaW5lciZ4PSU3QiUyMnclMjIlM0ElMjJUMFJUVUh4dGFXTnliM052Wm5RdWMyaGhjbVZ3YjJsdWRDMWtaaTVqYjIxOFlpRlZkM0kzTFMxSWEwSXdVMHhPYmtaMFVuTjNUbGg1TFdoc1IxOTJVMmhrU1hKVGJUUkVibGg2TWt4eFdrdE1Ubkl4Vm1aRVVrbFVWWGRUWjBjMk1IRmpmREF4UmpOV1ZFUkJNMU5LV2xJM05FSkxNelpHUWt4UVZETk1UamRMTkZCRU1sayUzRCUyMiUyQyUyMmklMjIlM0ElMjJmZmY5MDZmOS1lOGQ0LTRmYmEtODQ5OS02YzE3MTMzOTQ2NjIlMjIlN0Q%3D)   
{38F00546-B3FF-4D91-9798-BA6D79F8AEDE}
{1E5DB7EC-2091-40CF-A2D6-820310DBB5E9}

## Changes

- TableId will be accepted as datasource type specific property under Parameters
- Type will now have values either Table or Actions
- Associated unit tests

## Validation

- Local testing

### CODE REVIEW

File: src/Persistence/PaYaml/Models/SchemaV3/DataSourceInstance.cs [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
+    Table,
+    Actions
 }
-
 public record DataSourceInstance
 {
     [YamlMember(DefaultValuesHandling = DefaultValuesHandling.Preserve)]
-    public required DataSourceInstanceType Type { get; init; }
-    public string? TableLogicalName { get; init; }
+    public required DataSourceType Type { get; init; }
```
Comment (joem-msft) [inline src/Persistence/PaYaml/Models/SchemaV3/DataSourceInstance.cs:16]: this OM doesn't match the design in the description.
Namely, the `ConnectorId` property is missing. Also, it should be required. Do not create a special case for Dataverse.

File: src/schemas-tests/pa-yaml/v3.0/Examples/Src/DataSources/DataSources1.pa.yaml [hunk 2]
```diff
@@ -1,10 +1,13 @@
 DataSources:
```
Comment (joem-msft) [inline src/schemas-tests/pa-yaml/v3.0/Examples/Src/DataSources/DataSources1.pa.yaml:1]: not required for this PR, but do updatie the yaml schema file here to support native intellisense and validation support within VS Code. I can help you get started with this too. I've already added a simple validation for the DataSources top-level property.

see: https://github.com/microsoft/PowerApps-Tooling/blob/master/src/schemas/pa-yaml/v3.0/pa.schema.yaml

**Review by joem-msft — CHANGES_REQUESTED** [review]
The `ConnectorId` should be a required field. Using defaults that control schema type validation is not recommended for a flexible, scalable schema.

**Review by joem-msft — APPROVED** [review]

File: src/Persistence/PaYaml/Models/SchemaV3/DataSourceInstance.cs (hunk 1 above)
Comment (joem-msft) [reply src/Persistence/PaYaml/Models/SchemaV3/DataSourceInstance.cs:16]: I take back it needing to be required.

Comment (jamevaalet) [reply src/Persistence/PaYaml/Models/SchemaV3/DataSourceInstance.cs:16]: If ConnectorId is only required for non-dataverse datasources. If it's is missing, the default value will be dataverse.

**Pull request merged**